### PR TITLE
Move maven integration tests to dedicated job

### PR DESCRIPTION
**Pull request by geoand**
Similar to what we do with gradle integration tests

### CODE REVIEW

File: .github/workflows/ci-actions.yml [hunk 1]
```diff
@@ -110,7 +110,7 @@ jobs:
         java :
           - { name: Java8,
               java-version: 8,
-              maven_args: "-pl !integration-tests/gradle -pl !integration-tests/kubernetes/quarkus-standard-way"
+              maven_args: "-pl !integration-tests/gradle -pl !integration-tests/kubernetes/quarkus-standard-way -pl !integration-tests/kubernetes/maven"
```
Comment (jaikiran) [inline .github/workflows/ci-actions.yml:113]: Hello @geoand, I don't see `integration-tests/kubernetest/maven` in our repo. Did you mean `integration-tests/maven` instead?

Comment (geoand) [reply .github/workflows/ci-actions.yml]: Sorry, yeah :)

Comment (geoand) [reply .github/workflows/ci-actions.yml]: I changed the PR around a bit

**Comment by gsmet** [comment]
For the record, I think it's a good idea.

File: .github/workflows/ci-actions.yml [hunk 2]
```diff
@@ -272,6 +272,41 @@ jobs:
...
+    needs: build-jdk11
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        java:
+          - {
+            name: Java 11,
+            java-version: 11
+          }
```
Comment (gsmet) [inline .github/workflows/ci-actions.yml:287]: If we only test JDK 11, couldn't we reuse the artifacts as we do with the native tests?

**Comment by geoand** [comment]
It didn't seem to have much of an effect though...

File: .github/workflows/ci-actions.yml (hunk 2 above)
Comment (geoand) [reply .github/workflows/ci-actions.yml:287]: Yeah, I just did a quick and dirty test here. I'll update it tomorrow

**Comment by geoand** [comment]
To be merged after rebased onto https://github.com/quarkusio/quarkus/pull/11679

File: .github/workflows/ci-actions.yml (hunk 2 above)
Comment (geoand) [reply .github/workflows/ci-actions.yml:287]: Done

**Comment by geoand** [comment]
@gsmet can I get an approval for this now that https://github.com/quarkusio/quarkus/pull/11679 has been merged?

**Review by gsmet — APPROVED** [review]

**Pull request merged**